Source/SourceKittenFramework/library_wrapper.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  library_wrapper.swift
+//  sourcekitten
+//
+//  Created by Norio Nomura on 2/20/16.
+//  Copyright © 2016 SourceKitten. All rights reserved.
+//
+
+import Foundation
+
+struct DynamicLinkLibrary {
+    let path: String
+    let handle: UnsafeMutablePointer<Void>
+    
+    func loadSymbol<T>(symbol: String) -> T {
+        let sym = Darwin.dlsym(handle, symbol)
+        if sym == nil {
+            let errorString = String(UTF8String: dlerror())
+            fatalError("Finding symbol \(symbol) failed: \(errorString)")
+        }
+        return unsafeBitCast(sym, T.self)
+    }
+}
+
+let toolchainLoader = Loader(paths: [
+    xcodeDefaultToolchainOverride,
+    toolchainDir,
+    xcodeSelectPath?.toolchainDir,
+    /*
+    Followings are used when `xcode-select -p` points "Command Line Tools OS X for Xcode",
+    but Xcode.app exists.
+    */
+    applicationsDir?.xcodeDeveloperDir.toolchainDir,
+    applicationsDir?.xcodeBetaDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeBetaDeveloperDir.toolchainDir,
+    ].flatMap { path in
+        if let fullPath = path?.usrLibDir where fullPath.isFile {
+            return fullPath
+        }
+        return nil
+    })
+
+struct Loader {
+    let paths: [String]
+
+    func load(path: String) -> DynamicLinkLibrary {
+        let fullPaths = paths.map { $0.stringByAppendingPathComponent(path) }.filter { $0.isFile }
+
+        // try all fullPaths that contains target file
+        for fullPath in fullPaths {
+            let handle = dlopen(fullPath, RTLD_LAZY)
+            if handle != nil {
+                return DynamicLinkLibrary(path: path, handle: handle)
```

Comment:
That is used as:
```swift
private let library = toolchainLoader.load("sourcekitd.framework/Versions/A/sourcekitd")
private let library = toolchainLoader.load("libclang.dylib")
```
On those usage, path is identifier of dynamic link library.
e.g.
```sh
% otool -L /usr/local/Frameworks/SourceKittenFramework.framework/SourceKittenFramework
/usr/local/Frameworks/SourceKittenFramework.framework/SourceKittenFramework:
	/usr/local/opt/sourcekitten/Frameworks/SourceKittenFramework.framework/Versions/A/SourceKittenFramework (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libclang.dylib (compatibility version 1.0.0, current version 700.1.81)
	@rpath/sourcekitd.framework/Versions/A/sourcekitd (compatibility version 0.0.0, current version 700.1.81)
	@rpath/SWXMLHash.framework/Versions/A/SWXMLHash (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1256.14.0)
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1256.1.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
	@rpath/libswiftCore.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libswiftCoreGraphics.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libswiftDarwin.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libswiftDispatch.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libswiftFoundation.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libswiftObjectiveC.dylib (compatibility version 0.0.0, current version 0.0.0)
```
`@rpath/libclang.dylib` and `@rpath/sourcekitd.framework/Versions/A/sourcekitd` direct to DYLD as:
"search `libclang.dylib` from LC_RPATH"
"search `sourcekitd.framework/Versions/A/sourcekitd` from LC_RPATH"
I followed those DYLD's manner.